torch/ao/ns/fx/n_shadows_utils.py

```diff
@@ -904,6 +904,179 @@ def maybe_remap_node_to_shadow(node):
 
         model.recompile()
 
+def _get_weight_info_from_shadow_wrapper(shadow_wrapper: torch.nn.Module):
+    # input: shadow wrapper module
+    # output if shadow wrapper module has a weighted op:
+    #   (quantize_fn, (quantize_fn_args))
+    # output if shadow wrapper module doesn't have a weighted op:
+    #   None
+
+    # For now, assume that the weight is the second input
+    # to the shadow module. If that changes, we can fix it later.
+    placeholders_seen = 0
+    for shadow_n in shadow_wrapper.graph.nodes:  # type: ignore[union-attr]
+        if shadow_n.op != 'placeholder':
+            continue
+
+        placeholders_seen += 1
+        if placeholders_seen != 2:
+            continue
+
+        # the subgraph looks like
+        #
+        #   _input_scale_1 = self._input_scale_1
+        #   _input_zero_point_1 = self._input_zero_point_1
+        #   quantize_per_channel = torch.quantize_per_channel(
+        #       w2_0, _input_scale_1, _input_zero_point_1,
+        #       0, torch.qint8)
+        #
+        #  we have `w2_0`, and are navigating this subgraph
+        #  to get `_input_scale_1` and `_input_zero_point_1`
+
+        assert len(shadow_n.users) == 1
+        quant_node = list(shadow_n.users.keys())[0]
```

Comment:
nit: can we look for quant node and then get the weight node from quant_node.args[0], I feel traversing from node to its users is a bit hacky since there might be multiple users etc.

Reply:
I think that would work for the only case tested here, but it would be more hacky if there are multiple `quantize_per_tensor` functions.  Probably equally hacky to current approach?  I'm flexible, lmk.